src/test/java/org/folio/circulation/domain/policy/OverdueFinePolicyTest.java

```diff
@@ -0,0 +1,101 @@
+package org.folio.circulation.domain.policy;
+
+import static org.hamcrest.CoreMatchers.is;
+import static org.hamcrest.CoreMatchers.notNullValue;
+import static org.hamcrest.MatcherAssert.assertThat;
+
+import java.util.UUID;
+
+import org.junit.Before;
+import org.junit.Test;
+
+import io.vertx.core.json.JsonObject;
+
+public class OverdueFinePolicyTest {
+
+  private static JsonObject overdueFinePolicyJsonObject;
+
+  @Before
+  public void setUp() {
+    overdueFinePolicyJsonObject = new JsonObject()
+      .put("id", UUID.randomUUID().toString())
+      .put("name", "Overdue Fine Policy")
+      .put("overdueFine", new JsonObject()
+        .put("intervalId", "minute")
+        .put("quantity", 5.0))
+      .put("maxOverdueFine", 10.0)
+      .put("maxOverdueRecallFine", 10.0)
+      .put("gracePeriodRecall", true)
+      .put("countClosed", true);
+  }
+
+  @Test
+  public void shouldAcceptValidJsonRepresentation() {
+    OverdueFinePolicy overdueFinePolicy = OverdueFinePolicy.from(overdueFinePolicyJsonObject);
+
+    assertThat(overdueFinePolicy.getId(), is(overdueFinePolicyJsonObject.getString("id")));
+    assertThat(overdueFinePolicy.getName(), is(overdueFinePolicyJsonObject.getString("name")));
+    assertThat(overdueFinePolicy.getOverdueFineInterval(), is(OverdueFineInterval.fromValue(
+      overdueFinePolicyJsonObject.getJsonObject("overdueFine").getString("intervalId"))));
+    assertThat(overdueFinePolicy.getOverdueFine(),
+      is(overdueFinePolicyJsonObject.getJsonObject("overdueFine").getDouble("quantity")));
+    assertThat(overdueFinePolicy.getMaxOverdueFine(),
+      is(overdueFinePolicyJsonObject.getDouble("maxOverdueFine")));
+    assertThat(overdueFinePolicy.getMaxOverdueRecallFine(),
+      is(overdueFinePolicyJsonObject.getDouble("maxOverdueRecallFine")));
```

Comment:
That's maybe out if scope. 

Is the `quantity` is a double value not a real number? 
We should use `BigDecimal` for `maxOverdueFine` and `maxOverdueRecallFine`.

Reply:
I'm checking OverdueFinePolicy in mod-feesfines where it's defined. All of these fields (`quantity`, `maxOverdueFine`, `maxOverdueRecallFine`) are defined as `"type": "number"` which converts to Double in the automatically generated model. Do you propose to change it in both places?

Reply:
I think it is fine to change it in this class for now, but it would be good to have a tech-debt story for feesfines as well.

Reply:
Considering that model classes in mod-feesfines are generated automatically, what should we change the `"type": "number"` to in order for it to become BigDecimal in Java classes?

Reply:
I'm not sure, but you can try something like `"javaType": "java.lang.BigDecimal"`

Reply:
OK, thanks, I'll check the docs. I'll also update this PR (`quantity`, `maxOverdueFine`, `maxOverdueRecallFine` need to be BigDecimal, right?). 
Can you also point me to some kind of guide on how to choose numeric types? For examples, all these values are entered by user and basically represent money. Is there a clear set of rules that can help me decide whether I want to use Double or BigDecimal?

Reply:
> OK, thanks, I'll check the docs. I'll also update this PR (quantity, maxOverdueFine, maxOverdueRecallFine need to be BigDecimal, right?).

Yes, right. Not sure about `quantity`.

> Can you also point me to some kind of guide on how to choose numeric types? For examples, all these values are entered by user and basically represent money. Is there a clear set of rules that can help me decide whether I want to use Double or BigDecimal?

I'm not aware about such a doc. It is preferable to use BigDecimal for money, because Double has low accuracy when we sum/subtract/etc something. See this thread - https://stackoverflow.com/questions/3730019/why-not-use-double-or-float-to-represent-currency.

Reply:
`quantity` is how much we fine a patron per overdue period. For example, we can fine someone $2.5 per day.
I see, you mean this is a common sense practice rather than our procedural thing. I agree, BigDecimal would be safer.
But it seems like to change it to BigDecimal in this class I'll need to make a lot of changes in overdue fine calculator implementation and also it's tests and it really may be out of scope of this PR. What would you say if I create a technical debt issue for this?

Reply:
> What would you say if I create a technical debt issue for this?

I'm fine with this approach, thanks.

Reply:
> What would you say if I create a technical debt issue for this?

I'm fine with this approach, thanks.

Reply:
Leaving this link for reference [CIRC-655](https://issues.folio.org/browse/CIRC-655)